### PR TITLE
use StackExchange.Redis instead of ServiceStack.Redis package

### PR DESCRIPTION
**Pull request by RyanMarcotte**
As pointed out in #19 , the `ServiceStack.Redis` is a commercial product with free use caps and `StackExchange.Redis` is a free use library under MIT license.  I have made the library replacement.

### CODE REVIEW

File: src/Functional.CQS.AOP.Caching.Infrastructure.DistributedCache.Redis/ConnectionMultiplexerExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿using StackExchange.Redis;
+
+namespace Functional.CQS.AOP.Caching.Infrastructure.DistributedCache.Redis
+{
+	internal static class ConnectionMultiplexerExtensions
+	{
+		public static RedisClient GetRedisClient(this ConnectionMultiplexer source, FunctionalRedisCacheConfiguration configuration)
```
Comment (RyanMarcotte) [inline src/Functional.CQS.AOP.Caching.Infrastructure.DistributedCache.Redis/ConnectionMultiplexerExtensions.cs:10]: `StackExchange.Redis` [separates server-based and database-based commands](https://stackexchange.github.io/StackExchange.Redis/KeysScan).  The `RedisClient` object makes it easier for working with existing code.

File: src/Functional.CQS.AOP.Caching.Infrastructure.DistributedCache.Redis/RedisClientExtensions.cs [hunk 2]
```diff
@@ -78,7 +84,21 @@ internal static class RedisClientExtensions
 			});
 		}
 
-		private static Result<RedisText, Exception> ExecuteLuaShaSafely(this IRedisClient client, string sha1, string[] keys, string[] args) => Result.Try(() => client.ExecLuaSha(sha1, keys, args));
-		private static Result<IEnumerable<string>, Exception> ExecuteLuaShaAsListSafely(this IRedisClient client, string sha1, string[] keys, string[] args) => Result.Try(() => client.ExecLuaShaAsList(sha1, keys, args).AsEnumerable());
+		private static Result<RedisValue, Exception> ExecuteLuaShaSafely(this RedisClient client, LoadedLuaScript script, string[] keys, string[] args) => Result.Try(() => (RedisValue)client.ExecuteScript(script, keys, args));
+		private static Result<string[], Exception> ExecuteLuaShaAsListSafely(this RedisClient client, LoadedLuaScript script, string[] keys, string[] args) => Result.Try(() => (string[])client.ExecuteScript(script, keys, args));
+
+		private static T ExecCachedLua<T>(this RedisClient client, string script, Func<LoadedLuaScript, T> factory)
```
Comment (RyanMarcotte) [inline src/Functional.CQS.AOP.Caching.Infrastructure.DistributedCache.Redis/RedisClientExtensions.cs:93]: A new extension method (using a `ServiceStack.Redis` method name) so I did not have make as many changes to existing code.

File: src/Functional.CQS.AOP.Caching.Infrastructure.DistributedCache.Redis.Tests/FunctionalRedisCacheTests.cs [hunk 3]
```diff
@@ -485,6 +485,8 @@ public override bool Equals(object b)
 
 		#region Data
 
+		private const int CONCURRENT_REQUEST_COUNT = 5;
```
Comment (RyanMarcotte) [inline src/Functional.CQS.AOP.Caching.Infrastructure.DistributedCache.Redis.Tests/FunctionalRedisCacheTests.cs:488]: Used to be 25 concurrent requests...  The PR build was failing due to Redis timeouts, so I'll have to do some load testing / performance tuning.

**Review by JustinC-iQmetrix — APPROVED** [review]
This looks good to me! Thanks for responding so quickly to this request.

**Pull request merged**